bayes_opt/bayesian_optimization.py

```diff
@@ -87,7 +87,11 @@ def __init__(self, f, pbounds, random_state=None, verbose=2,
         self._verbose = verbose
         self._bounds_transformer = bounds_transformer
         if self._bounds_transformer:
-            self._bounds_transformer.initialize(self._space)
+            try:
+                self._bounds_transformer.initialize(self._space)
+            except (AttributeError, TypeError):
+                raise AssertionError('The transformer must be instance of'
```

Comment:
Is `AssertionError` the right choice? Maybe `TypeError` instead?

Reply:
You are right! `TypeError` is a better fit. Fixed.